app/forms/phone_verification_form.rb

```diff
@@ -4,9 +4,18 @@ class PhoneVerificationForm < QuestionsForm
   validates_presence_of :verification_code
 
   def valid?
-    return true if Rails.configuration.allow_magic_verification_code && verification_code == "000000"
-
     hashed_verification_code = VerificationCodeService.hash_verification_code_with_contact_info(@intake.sms_phone_number, verification_code)
+    # Magic codes provide a way of bypassing security in a development context.
+    # The easiest way to do this was to update the last entry to actually have the magic code.
+    if Rails.configuration.allow_magic_verification_code && verification_code == "000000"
+      token = TextMessageAccessToken.where(sms_phone_number: @intake.phone_number).last
+      if token.present?
+        token.update(
+          token: Devise.token_generator.digest(TextMessageAccessToken, :token, hashed_verification_code),
+          )
+      end
+      return true
+    end
```

Comment:
This code was prettymuch copied from the email_verification_form - except that code uses `EmailMessageAccessToken`